transactor/src/test/scala/io/mediachain/transactor/JournalCommitSpec.scala

```diff
@@ -0,0 +1,117 @@
+package io.mediachain.transactor
+
+import org.specs2.specification.{AfterAll, BeforeAll}
+
+import java.util.function.Consumer
+import java.util.concurrent.{BlockingQueue, LinkedBlockingQueue, TimeUnit}
+
+import io.atomix.copycat.client.CopycatClient
+import io.atomix.catalyst.transport.Address
+
+import Types._
+import StateMachine._
+
+object JournalCommitSpec extends io.mediachain.BaseSpec
+  with BeforeAll
+  with AfterAll
+{
+
+  def is =
+    sequential ^
+  s2"""
+  JournalStateMachine publishes commit events:
+   - when inserting an entity $insertEntity
+   - when extending the entity chain $extendEntityChain
+   - when extending the entity chain further $extendEntityChainFurther
+   - when inserting an artefact $insertArtefact
+   - when extending the artefact chain $extendArtefactChain
+  """
+
+  def beforeAll() {
+    JournalCommitSpecContext.setup()
+  }
+  
+  def afterAll() {
+    JournalCommitSpecContext.shutdown()
+  }
+  
+  def insertEntity = {
+    val context = JournalCommitSpecContext.context
+    val res = context.dummy.client.submit(JournalInsert(Entity(Map()))).join()
+    res.foreach((entry: CanonicalEntry) => {context.ref = entry.ref})
```

Comment:
Ok, I see `res` is an `Xor` and you're setting JournalCommitSpecContext's `ref` to that result and then adding to it in `extendEntityChain` etc. Can we express this w/o side effects, in more of a unit style? I think it's preferable to have each test set up its fixtures and test an action in isolation rather than statefully chaining them in this way. We can also separately have an integration test that runs all of the actions and checks that the final state is sane.